signals/generators/ios/ios_generator.py

```diff
@@ -39,41 +39,27 @@ def process(self):
             progress("Creating core data file")
             write_xml_to_file(self.core_data_path, self.schema.data_objects)
 
-        progress("Creating data model file")
-        self.create_header_file()
-        self.create_implementation_file()
-        self.copy_data_models()
-        self.check_setup_called()
-
-    def create_header_file(self):
-        self.process_template('data_model.h.j2', self.header_file, {})
+        if self.template_options[self.templates] == self.template_options['objc']:
+            progress('Preparing to generate Objective-C templates...')
+            template_to_generate = ObjectiveCTemplate(self.project_name,
+                                                      self.schema,
+                                                      self.data_models_path,
+                                                      self.jinja2_environment,
+                                                      BaseGenerator.BUILD_DIR)
+        else:
+            # Swift code goes here
+            progress('Preparing to generate Swift templates...')
+            template_to_generate = SwiftTemplate(self.project_name,
```

Comment:
Would the different templates potentially take different arguments in the future? If not you could set a variable to either `ObjectiveCTemplate` or `SwiftTemplate` then just instantiate the class from the variable.